e2e/nomostest/new.go

```diff
@@ -282,8 +282,9 @@ func FreshTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 
 	// Longer default wait timeout for stress tests
 	// Particularly the reset/finalizer takes a long time for the stress tests
+	// Current slowest test is TestStressLargeRequest
 	if *e2e.Stress {
-		nt.DefaultWaitTimeout = nt.DefaultWaitTimeout * 2
+		nt.DefaultWaitTimeout = 30 * time.Minute
```

Comment:
Increasing the reconcile timeout for the test might also reduce the total time it takes, since it won't have to re-actuate every 5m.

Reply:
The wait that needs it is in the `Reset` code that deletes the `RootSync` and [waits for `NotFound`](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/reset.go#L203), which is handled by the test scaffolding. To configure that per-test we would need another option on `NT` to set a specific value for that timeout. I figured it wasn't worth the extra option since this is a bit of an edge case and other waits in the stress tests need the longer default timeout

Reply:
I suppose we could set a different timeout for that wait condition if `*e2e.Stress`

Reply:
Note the stress test suite only runs a subset of ~4 targeted stress tests

Reply:
I guess this is fine for now. But perhaps we should refactor this in the future to add more granular timeouts so we don't get stuck in setup for 30m when something goes wrong.

Reply:
Sorry was working on another change but was meaning to make another iteration on this. Testgrid should be green after this change but I can make a followup change to make this more granular